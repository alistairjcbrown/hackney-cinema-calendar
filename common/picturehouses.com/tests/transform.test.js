const transform = require("../transform");
const moviesWithNoLocalScreenings = require("./mock-data/movie-with-no-local-screenings.json");
const moviesWithLocalScreenings = require("./mock-data/movie-with-local-screenings.json");

const formatParameter = (movies) => ({ movies });
const hackney = { domain: "https://www.picturehouses.com", cinemaId: "010" };
const finsburyPark = { domain: "https://www.picturehouses.com", cinemaId: "031" };

jest.useFakeTimers().setSystemTime(new Date("2024-08-01"));

describe("Picturehouse Cinema (common)", () => {
  describe("when no movies provided", () => {
    it("returns an empty list of events", async () => {
      const value = formatParameter([]);
      expect(await transform(hackney, value)).toEqual([]);
    });
  });

  describe("when movies with no local screenings provided", () => {
    it("returns an empty list of events", async () => {
      const value = formatParameter(moviesWithNoLocalScreenings);
      expect(await transform(hackney, value)).toEqual([]);
    });
  });

  describe("when movies with local screenings provided for Hackney", () => {
    it("returns an empty list of events", async () => {
      const value = formatParameter(moviesWithLocalScreenings);
      expect(await transform(hackney, value)).toMatchSnapshot();
    });
  });

  describe("when movies with local screenings provided for Finsbury Park", () => {
    it("returns an empty list of events", async () => {
      const value = formatParameter(moviesWithLocalScreenings);
      expect(await transform(finsburyPark, value)).toMatchSnapshot();
    });
  });
});
