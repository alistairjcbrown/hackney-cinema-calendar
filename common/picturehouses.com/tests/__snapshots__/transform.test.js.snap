// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picturehouse Cinema (common) when movies with local screenings provided for Finsbury Park returns an empty list of events 1`] = `
[
  {
    "overview": {
      "actors": [],
      "age-restriction": "U",
      "categories": [],
      "directors": [],
      "duration": 6060000,
      "trailer": "http://youtu.be/XRcOZZDvMv4",
    },
    "performances": [
      {
        "bookingUrl": "https://ticketing.picturehouses.com/Ticketing/visSelectTickets.aspx?cinemacode=031&txtSessionId=28204&visLang=1",
        "notes": "",
        "screen": "6",
        "time": 1722679200000,
      },
    ],
    "title": "Labyrinth",
    "url": "https://www.picturehouses.com/movie-details/031/HO00000223/Labyrinth",
  },
]
`;

exports[`Picturehouse Cinema (common) when movies with local screenings provided for Hackney returns an empty list of events 1`] = `
[
  {
    "overview": {
      "actors": [],
      "age-restriction": "U",
      "categories": [],
      "directors": [],
      "duration": 6060000,
      "trailer": "http://youtu.be/XRcOZZDvMv4",
    },
    "performances": [
      {
        "bookingUrl": "https://ticketing.picturehouses.com/Ticketing/visSelectTickets.aspx?cinemacode=010&txtSessionId=90342&visLang=1",
        "notes": "Kids' Club: Exclusively for children and their parents or carers. Unaccompanied adults are not admitted.",
        "screen": "4",
        "time": 1722687300000,
      },
      {
        "bookingUrl": "https://ticketing.picturehouses.com/Ticketing/visSelectTickets.aspx?cinemacode=010&txtSessionId=90364&visLang=1",
        "notes": "Kids' Club: Exclusively for children and their parents or carers. Unaccompanied adults are not admitted.",
        "screen": "1",
        "time": 1722766800000,
      },
    ],
    "title": "Labyrinth",
    "url": "https://www.picturehouses.com/movie-details/010/HO00000223/Labyrinth",
  },
]
`;
